src/pci.rs

```diff
@@ -319,6 +322,59 @@ impl PciDevice {
             );
         }
     }
+
+    pub fn allocate_bars(&mut self, start_address: Option<u64>) -> Option<u64> {
+        let mut next_address = start_address;
+
+        let mut current_bar_offset = 0x10;
+        let mut current_bar = 0;
+
+        //0x24 offset is last bar
+        while current_bar_offset <= 0x24 {
+            let bar = self.bars[current_bar];
+            if bar.size != 0 && bar.address == 0 {
+                if let Some(next_address) = next_address.as_mut() {
+                    match bar.bar_type {
+                        PciBarType::IoSpace | PciBarType::Unused => {}
+                        PciBarType::MemorySpace32 => {
+                            let address = ((*next_address + bar.size - 1) / bar.size) * bar.size;
```

Comment:
nit: you could write a small helper for calculating the aligned address.

Reply:
I just added it in response to your comment :-)